src/go/rpk/pkg/cli/transform/list.go

```diff
@@ -0,0 +1,152 @@
+* the Business Source License, use of this software will be governed
+* by the Apache License, Version 2.0
+ */
+
+package transform
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newListCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var detailed bool
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List data transforms",
+		Aliases: []string{"ls"},
+		Args:    cobra.NoArgs,
+		Run: func(cmd *cobra.Command, args []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]string{}); ok {
+				out.Exit(h)
+			}
+
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			api, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin api client: %v", err)
+
+			l, err := api.ListWasmTransforms(cmd.Context())
+			out.MaybeDie(err, "unable to list transforms: %v", err)
+
+			if detailed {
+				d := detailView(l)
+				printDetailed(f, d, os.Stdout)
+			} else {
+				s := summarizedView(l)
+				printSummary(f, s, os.Stdout)
+			}
+		},
+	}
+	cmd.Flags().BoolVarP(&detailed, "detailed", "d", false, "Print per-partition information for data transforms")
+	return cmd
+}
+
+type (
+	detailedTransformMetadata struct {
+		Name         string                              `json:"name"`
+		InputTopic   string                              `json:"input_topic"`
+		OutputTopics []string                            `json:"output_topics"`
+		Environment  map[string]string                   `json:"environment"`
+		Status       []adminapi.PartitionTransformStatus `json:"status"`
+	}
+	summarizedTransformMetadata struct {
+		Name         string            `json:"name"`
+		InputTopic   string            `json:"input_topic"`
+		OutputTopics []string          `json:"output_topics"`
+		Environment  map[string]string `json:"environment"`
+		Running      string            `json:"running"`
+	}
+)
+
+func makeEnvMap(env []adminapi.EnvironmentVariable) map[string]string {
+	out := make(map[string]string)
+	for _, entry := range env {
+		out[entry.Key] = entry.Value
+	}
+	return out
+}
+
+func summarizedView(metadata []adminapi.TransformMetadata) (resp []summarizedTransformMetadata) {
+	for _, meta := range metadata {
+		total := len(meta.Status)
+		running := 0
+		for _, v := range meta.Status {
+			if v.Status == "running" {
+				running++
+			}
+		}
+		resp = append(resp, summarizedTransformMetadata{
+			Name:         meta.Name,
+			InputTopic:   meta.InputTopic,
+			OutputTopics: meta.OutputTopics,
+			Environment:  makeEnvMap(meta.Environment),
+			Running:      fmt.Sprintf("%d / %d", running, total),
+		})
+	}
+	return
+}
+
+func printSummary(f config.OutFormatter, s []summarizedTransformMetadata, w io.Writer) {
```

Comment:
Nice seeing that you use an io.Writer here so it can be easily tested, thanks!

Reply:
Thanks (directly or indirectly) for making sure all these interfaces support io.Writer explicitly 😄